src/array/methods.jl

```diff
@@ -529,6 +529,14 @@ $message on dimension $D.
 To fix for `AbstractDimArray`, pass new lookup values as `cat(As...; dims=$D(newlookupvals))` keyword or `dims=$D()` for empty `NoLookup`.
 """
 
+function check_stack_dims(iter)
+    comparedims(Bool, dims.(iter)...; order=true, val=true, msg=Dimensions.Warn(" Can't `stack` AbstractDimArray, applying to `parent` object."))
+    iter
+end
+Base.stack(iter::AbstractArray{<:AbstractDimArray}; dims=:) = Base._stack(dims, check_stack_dims(iter))
```

Comment:
```suggestion
Base.stack(iter::AbstractArray{<:AbstractDimArray}; dims=:) = Base._stack(dimnum(first(iter), dims), check_stack_dims(iter))
```
    
This should allow passing Dimension/Symbol etc

Reply:
What about:
```julia
Base._stack(dims::Dimension, iter) = Base._stack(typeof(dims), iter)
Base._stack(dims::Type{<:Dimension}, iter) = Base._stack(dimnum(first(iter), dims), iter)
```
This helps because the default `dims` in Base's `Base.stack` is (oddly) `dims=:`. This has the same effect as `dims=ndims(first(iter))+1`, as far as I can tell, but it doesn't play well with `dimnums` in this context.

Reply:
Try to avoid dispatch on the underscore as much as possible... And we also need to accept Symbol. I would just define another method. That has dispatch for Colon and everything else uses `dimnum`